query/src/interpreters/interpreter_copy.rs

```diff
@@ -99,24 +100,22 @@ impl CopyInterpreter {
     // 3. Read from the stream and write to the table.
     // Note:
     //  We parse the `s3://` to ReadSourcePlan instead of to a SELECT plan is that:
-    //  COPY should deal with the file one by one and do some error handler on the OnError strategy.
-
-    #[tracing::instrument(level = "debug", name = "copy_one_file_to_table", skip(self), fields(ctx.id = self.ctx.get_id().as_str()))]
-    async fn copy_one_file_to_table(&self, file_name: Option<String>) -> Result<Vec<DataBlock>> {
+    #[tracing::instrument(level = "debug", name = "copy_files_to_table", skip(self), fields(ctx.id = self.ctx.get_id().as_str()))]
+    async fn copy_files_to_table(&self, files: Vec<String>) -> Result<Vec<DataBlock>> {
         let ctx = self.ctx.clone();
         let settings = self.ctx.get_settings();
 
-        let read_source_plan = self.plan.from.clone();
-        let read_source_plan = Self::rewrite_read_plan_file_name(read_source_plan, file_name);
-
-        tracing::info!("copy_one_file_to_table: source plan:{:?}", read_source_plan);
-
-        let from_plan = common_planners::SelectPlan {
-            input: Arc::new(PlanNode::ReadSource(read_source_plan)),
-        };
-
         let pipeline_builder = QueryPipelineBuilder::create(ctx.clone());
-        let mut pipeline = pipeline_builder.finalize(&from_plan)?;
+        let mut pipeline = NewPipeline::create();
+        for file in files {
+            let read_source_plan = self.plan.from.clone();
+            let read_source_plan = Self::rewrite_read_plan_file_name(read_source_plan, Some(file));
+            tracing::info!("copy_one_file_to_table: source plan:{:?}", read_source_plan);
+            let from_plan = common_planners::SelectPlan {
+                input: Arc::new(PlanNode::ReadSource(read_source_plan)),
+            };
+            pipeline = pipeline_builder.clone().finalize(&from_plan)?;
```

Comment:
For each file, we will build a new pipeline, but only take the last pipeline?

Reply:
Wrong code